src/Routing/Router.php

```diff
@@ -1042,7 +1042,12 @@ public static function parseRoutePath(string $url): array
                     if (!preg_match($paramsRegex, $param, $paramMatches)) {
                         throw new InvalidArgumentException("Could not parse a key=value from `{$param}`.");
                     }
-                    $convertedArray[$paramMatches['key']] = trim(trim($paramMatches['value'], '"'), "'");
+                    $paramKey = $paramMatches['key'];
+                    if (!preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $paramKey)) {
```

Comment:
Key is already checked and limited to `[a-z][a-z0-9-_]*`.

It same as placeholder regex 
https://github.com/cakephp/cakephp/blob/550fbc51919ab5cc6ff135814fee8cca831e52d0/src/Routing/Route/Route.php#L116

Reply:
The `params` match from the inital regex currently does match "invalid keys" aka invalid `key=value` pairs where the `key` can't be extracted to a PHP variable (thats how I understood Mark Story at least).
So `Bookmarks::view/-invalid=cakephp/1otherinvalid=cakephp` would result in a `params` of 
```
/-invalid=cakephp/1otherinvalid=cakephp
```
Which then gets pased down and needs to be re-evaluated.

Reply:
I also agree that having `value` data which itself contains `=` is a likely scenario.